src/platform/packages/shared/presentation/presentation_containers/interfaces/unsaved_changes/initialize_unsaved_changes.ts

```diff
@@ -46,16 +48,19 @@ export const initializeUnsavedChanges = <StateType extends object = object>({
   }
 
   const hasUnsavedChanges$ = anyStateChange$.pipe(
-    combineLatestWith(
-      parentApi.lastSavedStateForChild$(uuid).pipe(map((panelState) => panelState?.rawState))
-    ),
+    combineLatestWith(parentApi.lastSavedStateForChild$(uuid)),
     debounceTime(UNSAVED_CHANGES_DEBOUNCE),
     map(([, lastSavedState]) => {
-      const currentState = serializeState().rawState;
-      return !areComparatorsEqual(
+      const currentState = serializeState();
+
+      const equalRefs = checkRefEquality
+        ? isEqual(currentState.references, lastSavedState?.references)
+        : true;
```

Comment:
It seems from the failing tests, the vis embeddable in not correctly loading the references.

When you open a dashboard with an aggs-based vis, it's immediately dirty. If you attempt to **Reset**, nothing changes. If you refresh the browser from the dashboard, the refs are correctly loaded and the dirty state is clean.

But then going back to the dashboard list page, then back to the dashboard, the dirty state reappears.

https://github.com/user-attachments/assets/6fdfed21-5485-4bcf-93ff-c2f0ecb59131

So something is off with how we are loading the dashboard refs for the vis embeddable.

Reply:
Removing the controls from the example dashboard resolves the issue. Loading, reseting the dashboard works fine. So the controls are somehow interfering with the vis references.

Reply:
Ok I fixed this issue in f23ffb3.

First, we are missing the `<viz-panel-id>:kibanaSavedObjectMeta.searchSourceJSON.index` ref for the vis vis, so the initial refs was missing this and comparing it to the current vis with this ref loaded. If we save this dashboard, this ref would be correctly included. So I think this dashboard fixture json is just old and was never updated but since we never cared about the refs, outside of the panels rendering correctly, this never failed.

The second issue was that controls can impact the order of the `references$`. Since the `lodash.isEqual` fails if `_.isEqual(arr, arr.toReversed())`, the refs were all the same but out of order thus creating a dirty state. To fix this I just sort the refs before comparing.